### PR TITLE
fixes #2321 - remove new puppet creation option

### PR DESCRIPTION
**Pull request by ohadlevy**

### CODE REVIEW

**Comment by domcleal** [comment]
Should we do the same with environments?

**Comment by GregSutcliffe** [comment]
+1 for doing the same to environments, but should that be in this PR?

**Comment by ohadlevy** [comment]
hosts creation requires an environment, this will block non puppet users from using provisioning.

**Comment by domcleal** [comment]
Hm yeah, that's a shame.  One more question - the UI still allows a user to rename a Puppet class, should that be disabled?

**Comment by dLobatog** [comment]
I think so, reviewing the puppetclass model form would be good.

**Comment by ohadlevy** [comment]
I'm in favor of reviewing most of the old forms, but maybe in a seperate PR
?

**Comment by GregSutcliffe** [comment]
This doesn't seem to disable creation via the API:

```
$ curl -u admin:changeme -H "Accept: application/json" -H "content-type: application/json" -k http://127.0.0.1:3000/api/v2/puppetclasses -X POST -d '{"name":"foo"}'
{"created_at":"2014-09-09T15:37:19Z","global_class_params_count":0,"hostgroups_count":0,"hosts_count":0,"id":112,"lookup_keys_count":0,"name":"foo","updated_at":"2014-09-09T15:37:19Z"}
```

@ohadlevy I think you need to update the API controllers and routes as well.

**Comment by ohadlevy** [comment]
@GregSutcliffe I am not sure if we should remove the API, after all, IMHO if the user wants a way, we should allow it, removing it from the UI imho is enough?

**Comment by ohadlevy** [comment]
rebased and disabled name editing in the UI.

**Comment by dLobatog** [comment]
Why would we want to leave a way open? If we allow puppetclass creation through the API, we're allowing people to break the puppetclasses page. I think we shouldn't fix that as tons of variables can come with puppetclasses and creating a class manually is nearly always not worth the trouble.

However (@domcleal, @GregSutcliffe) please not do this to environments. The reason being importing environments does take a long time and adding a new environment is as easy as filling in the name. Think of this use case - user with 40 environments, has 41 environments actually in the master and wants its Foreman to be updated. The cronjob runs every hour and it takes very long to finish. In this case the user can just add manually the environment name, and click on the "import" button next to that environment. That should be quick and the user would save a lot of time.

**Comment by isratrade** [comment]
@ohadlevy, @eLobato, @domcleal, @GregSutcliffe, any decision on whether to remove ability to create puppet class using API. If yes, from both API v1 and v2?

**Comment by isratrade** [comment]
@ohadlevy, take this commit to failing fix integration test
https://github.com/isratrade/foreman/commit/d80457edaea4b25d7c35f98a7d7313fad0f57ca4

**Comment by isratrade** [comment]
integration test fixed and squashed on PR https://github.com/theforeman/foreman/pull/1822
If there are no more commits, then PR can be closed and PR #1822 can be merged after tests pass.

File: app/views/puppetclasses/_form.html.erb [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
   <div class="tab-content">
     <div class="tab-pane active" id="primary">
-      <%= text_f f, :name %>
+      <%= text_f f, :name, :disabled => true %>
```
Comment (isratrade) [inline app/views/puppetclasses/_form.html.erb:13]: @ohadlevy, this also means that puppet class cannot be renamed. If this is the intent, then I'll update the commit message to reflect this.

Comment (dLobatog) [inline app/views/puppetclasses/_form.html.erb:13]: I think this is fine.

**Pull request closed without merging**